doc/classes/Vector3.xml

```diff
@@ -306,9 +307,10 @@
 		</method>
 		<method name="reflect" qualifiers="const">
 			<return type="Vector3" />
-			<param index="0" name="n" type="Vector3" />
+			<param index="0" name="direction" type="Vector3" />
 			<description>
-				Returns the result of reflecting the vector from a plane defined by the given normal [param n].
+				Returns the result of reflecting the vector from a plane defined by the given direction vector [param direction].
+				[b]Note:[/b] [method reflect] differs from what other engines and frameworks call [code skip-lint]reflect()[/code]. In other engines, [code skip-lint]reflect()[/code] takes a normal direction which is a direction perpendicular to the plane. In Godot, you specify a direction parallel to the plane. See also [method bounce] which does what most engines call [code skip-lint]reflect()[/code].
```

Comment:
This doesn't make sense to me. You cannot uniquely define a plane using only a single parallel direction.

The C++ source code still calls the parameter `normal`:

```
Vector3 Vector3::reflect(const Vector3 &p_normal) const {
```

Same thing in 2D. I think this PR should be reverted, because it is not consistent with the C++ code, and for 3D it does not make any sense to define a plane by a parallel direction.

Reply:
@clayjohn ^

Reply:
As I mentioned in #89250, this PR should not be reverted. Instead, if the C++ code has the same inconsistency with all other standards like GLSL, it should be corrected as well instead of clinging to an old confusing and honestly simply needlessly nonstandard implementation

Reply:
@clayjohn There are an infinite amount of planes that pass through this parallel direction X axis, how does it determine which plane to use?

![plane](https://github.com/godotengine/godot/assets/1646875/4dee7af2-7a5d-4edd-9ffe-cc6eafd1b76a)

This PR does not fix the problem. The problem is that the bounce/reflect names are swapped from what users expect. It also introduces a new problem: not making any sense.

> Any plane aligned with this direction will result in the same reflected vector.

That doesn't make sense to me. Reflecting or bouncing over the planes in the GIF will produce different results.

> The description is consistent with the C++ code, maybe not with the parameter name, but again.

Yes, that's what I mean. The parameter names are not consistent.

> I am not going to support incorrect documentation just to be internally consistent.

But now it's neither correct nor consistent.

Reply:
> @clayjohn There are an infinite amount of planes that pass through this parallel direction X axis, how does it determine which plane to use?

You are right. I was oversimplifying above and can see that now. 

When visualizing the direction as a vector across a given plane (with a specific normal) it makes sense. But indeed, for any direction there are an infinite number of possible planes, so you would need the specific normal to know which plane is being used. 

For 2D, it works because a line can be defined by a given direction, but that doesn't extend to 3D. 

I'll make a PR with a better description!